db.go

```diff
@@ -750,6 +750,16 @@ func (db *DB) Snapshot(dir string, withHead bool) error {
 // Querier returns a new querier over the data partition for the given time range.
 // A goroutine must not handle more than one open Querier.
 func (db *DB) Querier(mint, maxt int64) (Querier, error) {
+	// Substract the retention from the current time and convert it to milliseconds so we can compare it with mint, maxt.
+	retentionCutoff := time.Now().Add(-time.Duration(db.opts.RetentionDuration)*time.Millisecond).UnixNano() / int64(time.Millisecond)
+	// Limit the querer to the retention time.
+	if mint < retentionCutoff {
+		mint = retentionCutoff
+	}
+	if maxt < retentionCutoff {
```

Comment:
Can we return an error in this case then? or immediately return the empty result? If you ask the data before retention.

Reply:
Querying data that is not present is never an error, so empty result would be the right thing.

Reply:
Yea, The issue is that code could potentially still return single sample even when retention is `T-10` and we ask for `T-1000 - T-100`, because we will overwrite mint and maxt with `T-10`

Reply:
Yep, good point, so we should just return an empty result here (but not an error).

Reply:
(Well, we have to return a querier, but then it should be one that doesn't produce any data.)